pkg/operator/controller/dns/controller.go

```diff
@@ -121,7 +121,18 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	if dnsConfig.Spec.PublicZone != nil {
 		zones = append(zones, *dnsConfig.Spec.PublicZone)
 	}
+	statuses, result := r.publishRecordToZones(zones, record)
+	// TODO: only update if status changed
+	updated := record.DeepCopy()
+	updated.Status.Zones = statuses
```

Comment:
It occurs to me that we should be using some kind of `mergeConditions` function here, but that is a separate issue, and we can do it in a follow-up PR.

Reply:
OK